### DISCUSSION TITLE
Project Chrono vehicle doesn't drive

### DISCUSSION THREAD

**Discussion by Makro007** [carla-simulator/carla#4921] [Q&A] [answered]
Hi everyone. I am trying to test project chrono in Carla and after following the steps described in the "Chrono Integration" doc I find myself stuck with a non moving car. I followed all the steps including launch with chrono enabled and I start it with the python example "manual_control_chrono.py". Does anyone have an idea why the chrono vehicle wouldn't move? 
Checked in DEBUG mode and the TICK component is never called. Thanks for any tips/help!

![image](https://user-images.githubusercontent.com/15676343/142985924-7cf7745e-8b73-4f72-ac4d-d8c4d83f0e54.png)

**Comment by Makro007** [comment, marked as answer]
Found the issue. OnVehicleOverlap is hitting in the initialization so it always disables chrono physics in the start.

**Reply by j-scharrenbach** [reply]
Hi Makro007,

I am having the same issue as you described above.
Whenever I activate chrono on a vehicle, immediately a warning is printed:

```
WARNING: Chrono physics does not support collisions yet, reverting to default PhysX physics.
```

Do you know any solution on how to deal with this? Have you somehow managed to get chrono working properly without stopping due to the collision?

Big thanks in advance,
Jannik

**Reply by Makro007** [reply]
Hi Jahnik.

Yes in my case I added my own collision handling. Basically I send the information about the collision to the chrono vehicle and simulate the force. It works nicely with other vehicles but it doesn't work yet with the terrain overlapping part. I can send you my example if it would help?

**Reply by j-scharrenbach** [reply]
Hi,

sounds really interesting, thank you for the fast response.

I would highly appreciate if you could send me your implementation. It would help me a lot.
What would be the easiest way of sending it to me?

Thanks a lot!
Jannik

**Reply by Makro007** [reply]
Hi,
I just zipped the 2 files, there's some other parts of code that wont be of help to you but the main is everything connected to the variable "std::shared_ptr<chrono::ChForce> hitForce". If you'll have any problems let me know :)  

[MovementComponents.zip](https://github.com/carla-simulator/carla/files/7769254/MovementComponents.zip)

**Reply by j-scharrenbach** [reply]
Thank you so much! I will take a look at it after christmas and come back if I have any questions :)

**Reply by rohan-khaire25** [reply]
@Makro007 Hello, I am having trouble making this work. Could you tell me how to go about it? Thank you.

**Reply by Makro007** [reply]
@rohan-khaire25 Hi. Where is the issue? Did you try to implement the code example?

**Reply by rohan-khaire25** [reply]
@Makro007 Thank you for your reply. I am sorry I don't know how to implement this code. Could you please help me out with it?

**Reply by Makro007** [reply]
@rohan-khaire25 If you are on the [carla/axel/chrono_implementation](https://github.com/carla-simulator/carla/tree/axel/chrono_implementation) branch, then you already have all of the implementation. This zipped files are just an upgrade to implement collisions with chrono instead of stopping the chrono vehicle on collision.

**Reply by rohan-khaire25** [reply]
@Makro007 Thank you so much. i will use the said branch and try to run chrono with the modification.

**Reply by Himverma2012** [reply]
> @rohan-khaire25 If you are on the [carla/axel/chrono_implementation](https://github.com/carla-simulator/carla/tree/axel/chrono_implementation) branch, then you already have all of the implementation. This zipped files are just an upgrade to implement collisions with chrono instead of stopping the chrono vehicle on collision.

Hello, Do we have to build the carla again with this zip file?

**Reply by Himverma2012** [reply]
I tried to build chrono using the https://github.com/carla-simulator/carla/tree/axel/chrono_implementation, but the build failed with Error-6

**Reply by Makro007** [reply]
@Himverma2012 Hey. I didn't see any updates on the branch so probably it should still work fine. Maybe check what this error means and fix it. If you can build the chrono lib then you just have to follow the Carla instructions on how to use chrono [chrono](https://carla.readthedocs.io/en/latest/tuto_G_chrono/) and paste the code from the zip to Carla project. After building the carla project it should work :)
